python/rascaline/rascaline/utils/splines.py

```diff
@@ -0,0 +1,396 @@
+            self.spline_derivatives = concatenated_derivatives[sort_indices]
+
+        return self.spline_positions, self.spline_values, self.spline_derivatives
+
+    def _compute_from_spline(self, positions):
+        x = positions
+        delta_x = self.spline_positions[1] - self.spline_positions[0]
+        n = (np.floor(x / delta_x)).astype(np.int32)
+
+        t = (x - n * delta_x) / delta_x
+        t_2 = t**2
+        t_3 = t**3
+
+        h00 = 2.0 * t_3 - 3.0 * t_2 + 1.0
+        h10 = t_3 - 2.0 * t_2 + t
+        h01 = -2.0 * t_3 + 3.0 * t_2
+        h11 = t_3 - t_2
+
+        p_k = self.spline_values[n]
+        p_k_1 = self.spline_values[n + 1]
+
+        m_k = self.spline_derivatives[n]
+        m_k_1 = self.spline_derivatives[n + 1]
+
+        new_shape = (-1,) + (1,) * self.number_of_custom_axes
+        h00 = h00.reshape(new_shape)
+        h10 = h10.reshape(new_shape)
+        h01 = h01.reshape(new_shape)
+        h11 = h11.reshape(new_shape)
+
+        interpolated_values = (
+            h00 * p_k + h10 * delta_x * m_k + h01 * p_k_1 + h11 * delta_x * m_k_1
+        )
+
+        return interpolated_values
+
+
+class SplinerFromFunction(SplinerBase):
+    """Compute the radial integral spline points based on a provided function.
+
+    :parameter radial_integral: Function to compute the radial integral. Function must
+        take ``n``, ``l``, and ``positions`` as inputs, where ``n`` and ``l`` are
+        integers and ``positions`` is a numpy 1-D array that contains the spline points
+        at which the radial integral will be evaluated. The function must return a numpy
+        1-D array containing the values of the radial integral. :parameter cutoff:
+        Spherical cutoff for the radial basis :parameter max_angular: number of radial
+        components
+    :parameter max_angular: number of radial components
+    :parameter max_radial: number of angular components
+    :parameter spline_cutoff: cutoff radius for the spline interpolation. This is also
+        the maximal value that can be interpolated.
+    :parameter accuracy: accuracy of the numerical integration and the splining.
+        Accuracy is reached when either the mean absolute error or the mean relative
+        error gets below the ``accuracy`` threshold.
+    :parameter radial_integral_derivative: The derivative of the radial integral taking
+        the same paramaters as ``radial_integral``. If it is :py:obj:`None` (default),
+        finite differences are used to calculate the derivative of the radial integral.
+        It is recommended to provide this parameter if possible. Derivatives from finite
+        differences can cause problems when evaluating at the edges of the domain (i.e.,
+        at ``0`` and ``spline_cutoff``) because the function might not be defined
+        outside of the domain.
+    :parameter center_contribution: Contribution of the central atom required for LODE
+        calculations.
```

Comment:
This should point to the corresponding property documentation in the base class

Reply:
The properties are private. I don't know what we should point to.